packages/@lwc/ssr-compiler/src/transmogrify.ts

```diff
@@ -0,0 +1,179 @@
+    mode: TransmogrificationMode;
+}
+
+export type Visitors = Parameters<typeof traverse<Node, TransmogrificationState>>[1];
+
+const EMIT_IDENT = b.identifier('$$emit');
+// Rollup may rename variables to prevent shadowing. When it does, it uses the format `foo$0`, `foo$1`, etc.
+const TMPL_FN_PATTERN = /tmpl($\d+)?/;
+const GEN_MARKUP_PATTERN = /generateMarkup($\d+)?/;
+
+const isWithinFn = (pattern: RegExp, nodePath: NodePath): boolean => {
+    const { node } = nodePath;
+    if (!node) {
+        return false;
+    }
+    if (node.type === 'FunctionDeclaration' && pattern.test(node.id.name)) {
+        return true;
+    }
+    if (nodePath.parentPath) {
+        return isWithinFn(pattern, nodePath.parentPath);
+    }
+    return false;
+};
+
+const visitors: Visitors = {
+    FunctionDeclaration(path, state) {
+        const { node } = path;
+        if (!node?.async || !node?.generator) {
+            return;
+        }
+
+        // Component authors might conceivably use async generator functions in their own code. Therefore,
+        // when traversing & transforming written+generated code, we need to disambiguate generated async
+        // generator functions from those that were written by the component author.
+        if (!isWithinFn(GEN_MARKUP_PATTERN, path) && !isWithinFn(TMPL_FN_PATTERN, path)) {
+            return;
+        }
+        node.generator = false;
+        node.async = state.mode === 'async';
+        node.params.unshift(EMIT_IDENT);
+    },
+    YieldExpression(path, state) {
+        const { node } = path;
+        if (!node) {
+            return;
+        }
+
+        // Component authors might conceivably use generator functions within their own code. Therefore,
+        // when traversing & transforming written+generated code, we need to disambiguate generated yield
+        // expressions from those that were written by the component author.
+        if (!isWithinFn(TMPL_FN_PATTERN, path) && !isWithinFn(GEN_MARKUP_PATTERN, path)) {
+            return;
+        }
+
+        if (node.delegate) {
+            // transform `yield* foo(arg)` into `foo($$emit, arg)` or `await foo($$emit, arg)`
+            if (node.argument?.type !== 'CallExpression') {
+                throw new Error(
+                    'Implementation error: cannot transmogrify complex yield-from expressions'
+                );
+            }
+
+            const callExpr = node.argument;
+            callExpr.arguments.unshift(EMIT_IDENT);
+
+            if (state.mode === 'sync') {
+                path.replaceWith(callExpr);
+            } else {
+                path.replaceWith(b.awaitExpression(callExpr));
+            }
+        } else {
+            // transform `yield foo` into `$$emit(foo)`
+            const emittedExpression = node.argument;
+            if (!emittedExpression) {
+                throw new Error(
+                    'Implementation error: cannot transform a yield expression that yields nothing'
```

Comment:
Why not? Bare `yield` is the same as `yield undefined`, and we're not doing any other special handling.

Reply:
It is not that `yield` or `yield undefined` is invalid code. It is that the transmogrifier explicitly doesn't handle this situation. I don't think we need it anyway, because `yield undefined` would result in `markup += undefined` which would append the string `'undefined'` to the generated markup (i.e. `"" = undefined === 'undefined'`), which almost certainly would never be an intentional choice.